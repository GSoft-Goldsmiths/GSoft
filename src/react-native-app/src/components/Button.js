import React from 'react';
import { style } from '../styles/variables';
import { Text, TouchableHighlight } from 'react-native';

export default ({ text, handler }) => (
  <TouchableHighlight
    style={style.button}
    onPress={() => handler()}
  >
    <Text style={{ color: '#fff' }}>{text}</Text>
  </TouchableHighlight>
);