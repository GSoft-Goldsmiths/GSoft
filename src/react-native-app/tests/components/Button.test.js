import React from 'react';
import { Text, TouchableHighlight } from 'react-native';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

import Button from '../../src/components/Button';

configure({ adapter: new Adapter() });


describe('<Button />', ()=>{
  it('renders without crashing', () => {
    const rendered = renderer.create(<Button text="title" handler={() => {}}/>).
      toJSON();
    expect(rendered).toMatchSnapshot();
  });
  
  it('renders with correct elements', () => {
    const wrapper = shallow(<Button text="title" handler={() => {}}/>);
    expect(wrapper.find(Text).exists()).toBeTruthy();
    expect(wrapper.find(TouchableHighlight).exists()).toBeTruthy();
  });

  it('renders with correct props', () => {
    const wrapper = shallow(<Button text="title" handler={() => { testData =  "button is pressed"; }}/>);
    const text = wrapper.find(Text);
    expect(text.props().children).toEqual("title");
  });

  it('the button handler should work if the button is pressed', () => {
    let testData = "before button is pressed";
    const wrapper = shallow(<Button text="title" handler={() => { testData =  "button is pressed"; }}/>);
    wrapper.find(TouchableHighlight).simulate('press');
    expect(testData).toEqual("button is pressed");
  });
});

