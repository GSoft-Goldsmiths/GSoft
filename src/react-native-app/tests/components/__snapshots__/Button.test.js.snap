// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders without crashing 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0093A8",
      "height": 36,
      "justifyContent": "center",
      "marginVertical": 20,
      "width": 90,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Caviar Dreams Bold",
      }
    }
  >
    title
  </Text>
</View>
`;
